validation/util/test.go

```diff
@@ -184,3 +248,74 @@ func RuntimeOutsideValidate(g *generate.Generator, f AfterFunc) error {
 	}
 	return nil
 }
+
+// RuntimeLifecycleValidate validates runtime lifecycle.
+func RuntimeLifecycleValidate(g *generate.Generator, lifecycle *LifecycleFuncs, actions LifecycleAction) error {
+	bundleDir, err := PrepareBundle()
+	if err != nil {
+		return err
+	}
+	r, err := NewRuntime(RuntimeCommand, bundleDir)
+	if err != nil {
+		os.RemoveAll(bundleDir)
+		return err
+	}
+	defer r.Clean(true, true)
+	err = r.SetConfig(g)
+	if err != nil {
+		return err
+	}
+	r.SetID(uuid.NewV4().String())
+
+	if lifecycle != nil && lifecycle.Setup != nil {
+		if err := lifecycle.Setup(&r); err != nil {
+			return err
+		}
+	}
+
+	if actions&LifecycleActionCreate != 0 {
+		stderr, err := r.Create()
+		if err != nil {
+			os.Stderr.WriteString("failed to create the container\n")
+			os.Stderr.Write(stderr)
+			return err
+		}
+	}
+
+	if lifecycle != nil && lifecycle.PostCreate != nil {
+		if err := lifecycle.PostCreate(&r); err != nil {
+			return err
+		}
+	}
+
+	if actions&LifecycleActionStart != 0 {
+		stderr, err := r.Start()
+		if err != nil {
+			os.Stderr.WriteString("failed to start the container\n")
+			os.Stderr.Write(stderr)
+			return err
+		}
+	}
+
+	if lifecycle != nil && lifecycle.PreStop != nil {
+		if err := lifecycle.PreStop(&r); err != nil {
+			return err
+		}
+	}
+
+	if actions&LifecycleActionDelete != 0 {
+		stderr, err := r.Delete()
+		if err != nil {
+			os.Stderr.WriteString("failed to delete the container\n")
+			os.Stderr.Write(stderr)
+			return err
+		}
+	}
+
+	if lifecycle != nil && lifecycle.PostStop != nil {
```

Comment:
I think anything running after `Delete` should be `PostDelete`.  This PR currently seems to handle `stop` and `delete` in one step (or maybe it is never intended to call `stop` and instead waits for the container to stop on its own?), but calling this setting `PostDelete` makes sense regardless of an explicit stop or not.

Reply:
So all the embedded functions should have different new, I'll update it too.

I prefer calling the 'stop' in the related test cases, for example, if we want to test 'kill'.
We can add test codes in the 'PreDelete'.   (PreDelete is better than PreStop)